src/virtualenv/util/lock.py

```diff
@@ -113,8 +139,30 @@ def lock_for_key(self, name, no_block=False):
         finally:
             self._del_lock(lock)
 
+    @contextmanager
+    def non_reentrant_lock_for_key(self, name):
+        with _CountedFileLock(str(self.path / "{}.lock".format(name))):
+            yield
+
+
+class NoOpFileLock(FileLockBase):
```

Comment:
Calling this file lock feels a bit cheating, maybe we want to move the interface to a simple Lock, and then inherit this from that?

Reply:
isn't that what this does?

Reply:
What I'm saying is:

```python
class Lock(ABC):
     pass

class FileLock(Lock):
     pass

class NoOpLock(Lock):
   pass
```

Reply:
I don't see how renaming `FileLockBase` improves anything but I'll do it if you want

Reply:
Yeah, let's do that please.